axelor-account/src/main/java/com/axelor/apps/account/service/batch/BatchBlockCustomersWithLatePayments.java

```diff
@@ -0,0 +1,120 @@
+package com.axelor.apps.account.service.batch;
+
+import com.axelor.apps.account.db.AccountConfig;
+import com.axelor.apps.account.db.Invoice;
+import com.axelor.apps.account.db.repo.InvoiceRepository;
+import com.axelor.apps.account.service.config.AccountConfigService;
+import com.axelor.apps.base.db.Partner;
+import com.axelor.apps.base.service.app.AppBaseService;
+import com.axelor.db.JPA;
+import com.axelor.db.Query;
+import com.axelor.exception.AxelorException;
+import com.axelor.exception.db.repo.ExceptionOriginRepository;
+import com.axelor.exception.service.TraceBackService;
+import com.axelor.i18n.I18n;
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.collections.CollectionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BatchBlockCustomersWithLatePayments extends BatchStrategy {
+  protected final Logger log = LoggerFactory.getLogger(getClass());
+
+  private AccountConfigService accountConfigService;
+  private InvoiceRepository invoiceRepository;
+  private AppBaseService appBaseService;
+
+  @Inject
+  public BatchBlockCustomersWithLatePayments(
+      AccountConfigService accountConfigService,
+      InvoiceRepository invoiceRepository,
+      AppBaseService appBaseService) {
+    this.accountConfigService = accountConfigService;
+    this.invoiceRepository = invoiceRepository;
+    this.appBaseService = appBaseService;
+  }
+
+  @Override
+  protected void process() {
+    try {
+      String result = blockCustomersWithLatePayments();
+      addComment(result);
+    } catch (Exception e) {
+      TraceBackService.trace(e, ExceptionOriginRepository.IMPORT, batch.getId());
+      incrementAnomaly();
+    }
+  }
+
+  protected String blockCustomersWithLatePayments() throws AxelorException {
+    String result = "";
+    List<Invoice> invoices;
+    List<Long> customersToBlock = new ArrayList<Long>();
+    int offset = 0;
+    Query<Invoice> query =
+        invoiceRepository
+            .all()
+            .filter(
+                "self.operationTypeSelect = :operationTypeSelect "
+                    + "AND self.amountRemaining > 0");
+    while (!(invoices =
+            query
+                .bind("operationTypeSelect", InvoiceRepository.OPERATION_TYPE_CLIENT_SALE)
+                .fetch(FETCH_LIMIT, offset))
+        .isEmpty()) {
+      for (Invoice invoice : invoices) {
+        ++offset;
+        Partner partner = processInvoice(invoice);
+        if (partner != null) {
+          if (!customersToBlock.contains(partner.getId())) {
+            log.debug("Blocking {}", partner.getFullName());
+            result += I18n.get("Blocking") + " " + partner.getFullName() + "</br>";
+            customersToBlock.add(partner.getId());
+          }
+        }
```

Comment:
the `try {...} catch` must catch any exception happening inside the for loop.
* the process must call `incrementDone()` at the end of the loop (when no exceptions occured)
* the process must call `incrementAnomaly()` when an exception happens.